src/stan-users-guide/parallel-computing.Rmd

```diff
@@ -1,12 +1,252 @@
-# Map-Reduce  {#map-reduce.chapter}
+# Parallel Computing  {#parallel-computing.chapter}
+
+Stan has two mechanisms for parallelizing calculations used in a model: `reduce_sum` and and `map_rect`.
+
+The main advantages to `reduce_sum` are:
+
+1. `reduce_sum` has a more flexible argument interface, avoiding the packing and unpacking that is necessary with `map_rect`.
+2. `reduce_sum` partitions the data for parallelization automatically (this is done manually in `map_rect`).
+3. `reduce_sum` is easier to use.
+
+while the advantages of `map_rect` are:
+
+1. `map_rect` returns a list of vectors, while `reduce_sum` returns only a real.
+2. `map_rect` can be parallelized across multiple computers, while `reduce_sum` can only parallelized across multiple cores.
+
+## Reduce-Sum { #reduce-sum }
+
+```reduce_sum``` is a tool for parallelizing operations that can be represented as a sum of functions, `g: U -> real`.
+
+For instance, for a sequence of ```x``` values of type ```U```, ```{ x1, x2, ... }```, we might compute the sum:
+
+```g(x1) + g(x2) + ...```
+
+In probabilistic modeling this comes up when there are N conditionally independent terms in a likelihood. Because of the conditional independence, these terms can be computed in parallel. If dependencies exist between the terms, then this isn't possible. For instance, in evaluating the log density of a Gaussian process ```reduce_sum``` would not be very useful.
+
+```reduce_sum``` doesn't actually take ```g: U -> real``` as an input argument. Instead it takes ```f: U[] -> real```, where ```f``` computes the partial sum corresponding to the slice of the sequence ```x``` passed in. For instance:
+
+```
+f({ x1, x2, x3 }) = g(x1) + g(x2) + g(x3)
+f({ x1 }) = g(x1)
+f({ x1, x2, x3 }) = f({ x1, x2 }) + f({ x3 })
+```
+
+If the user can write a function ```f: U[] -> real``` to compute the necessary partial sums in the calculation, then we can provide a function to automatically parallelize the calculations (and this is what ```reduce_sum``` is).
```

Comment:
flip this around? - the reduce_sum function takes the user-specified function `f: U[] -> real` and automatically parallelizes these calculations.